plugins/pagerduty/komand_pagerduty/util/api.py

```diff
@@ -1,51 +1,205 @@
+        """
+        return self.send_request(method="GET", path=f"/users/{user_id}/")
+
+    def list_users(self) -> dict:
+        """
+        Formats the request to get a list of all of the users in pagerduty
+
+
+        :return dict: This will return a list of user objects
+        """
+        return self.send_request(method="GET", path="/users/")
+
+    def send_request(
+        self, method: str, path: str, params: dict = None, payload: dict = None, headers: dict = None, data: dict = None
+    ) -> Union[dict, bool]:
+        """
+        A wrapper with error handling for making requests to the pager duty api
+
+        :param str method: The type of request that will be made
+        :param str path: The path that will be appended to the base url
+        :param dict params: Any paramaters that will be added to the request, defaults to None
+        :param dict payload: Any data that will be added to the json section of the request, defaults to None
+        :param dict headers: Any additional headers that will be added to the request, defaults to None
+        :param dict data: Any data that will be added to the data section of the request, defaults to None
+        :raises PluginException: If there is an error connecting to the pager duty api or else if there is no valid data returned
+        :return Union[dict, bool]:
+                dict: The response of the request in json format
+                bool: If the delete request is called then there will be no data returned from the api, so we return True
+        """
+
+        if not headers:
+            headers = self.headers
 
-    def send_request(self, method: str, path: str, params: dict = None, payload: dict = None) -> dict:
         try:
             response = self.session.request(
                 method.upper(),
                 "https://api.pagerduty.com" + path,
                 params=params,
                 json=payload,
-                headers=self.headers,
+                headers=headers,
+                data=data,
             )
 
-            if response.status_code == 401:
-                raise PluginException(preset=PluginException.Preset.USERNAME_PASSWORD, data=response.text)
-            if response.status_code == 403:
-                raise PluginException(preset=PluginException.Preset.API_KEY, data=response.text)
-            if response.status_code == 404:
-                raise PluginException(preset=PluginException.Preset.NOT_FOUND, data=response.text)
-            if 400 <= response.status_code < 500:
-                raise PluginException(
-                    preset=PluginException.Preset.UNKNOWN,
-                    data=response.text,
-                )
-            if response.status_code >= 500:
-                raise PluginException(preset=PluginException.Preset.SERVER_ERROR, data=response.text)
-
+            if method.upper() == "DELETE" and response.status_code == 204:
+                return True
```

Comment:
How we normally handle `204`s is something like this:
```
if response.status_code==204:
     return True/None
if 200 < response.status_code < 300:
     return normally
```
So we simply catch 204 first because it's always the result of delete, otherwise continue to catch 200s and return the regular json.

Reply:
This has been updated to match the above format